<launch>
  <!-- "camera" should uniquely identify the device. All topics are pushed down
       into the "camera" namespace, and it is prepended to tf frame ids. -->
  <arg name="camera" default="XTION" />
  <arg name="rgb_frame"   default="/$(arg camera)_RGB" />
  <arg name="depth_frame" default="/$(arg camera)_IR" />
  <arg name="calibration_url"  default="$(find arm_robot_model)/calib/xtion_calib.yaml" />
  <arg name="point_cloud_topic" default="/XTION/depth/points" />
  <arg name="camera_info_topic" default="/XTION/depth/camera_info" />
  
  <node pkg="oni_vicon_player" name="$(arg camera)" type="oni_vicon_player_node" respawn="false" output="screen">
    <param name="rgb_frame_id"  type="string"  value="$(arg rgb_frame)" />
    <param name="depth_frame_id"  type="string"  value="$(arg depth_frame)" />
    <param name="calib_url"  type="string"  value="$(arg calibration_url)" />   
	<param name="point_cloud_topic"    type="string"  value="$(arg point_cloud_topic)" />
	<param name="camera_info_topic"  type="string"  value="$(arg camera_info_topic)" />
  </node>
  
  
  <!-- Load reasonable defaults for the relative pose between cameras -->
  <include file="$(find oni_vicon_player)/launch/arm_rgbd_frames.launch">
    <param name="camera" type="string" value="/$(arg camera)"/>
  </include>
</launch>
